src/main/java/io/confluent/kafkarest/SimpleConsumerManager.java

```diff
@@ -124,157 +90,101 @@ public void consume(final String topicName,
                       final EmbeddedFormat embeddedFormat,
                       final ConsumerManager.ReadCallback callback) {
 
-    List<ConsumerRecord> records = null;
+    List<ConsumerRecord> records = new ArrayList<>();
```

Comment:
A minor optimization would be to move this into the `else` branch below so that we can allocate with the exact number of records. Otherwise we can use `Collections.emptyList()`.